vllm/model_executor/models/plamo2.py

```diff
@@ -0,0 +1,868 @@
+        capacity_factor: float = 1.0,
+        group_size: int = 1024,
+        sparse_step: Optional[int] = None,
+        sparse_intermediate_size: Optional[int] = None,
+        shared_intermediate_size: Optional[int] = None,
+        # FP8
+        linear_type: LinearType = LinearType.Normal,
+        fp8_accum_dtype: Optional[str] = None,
+        # Evaluation
+        eval_attention_n_bit: Optional[int] = None,
+        eval_mlp_n_bit: Optional[int] = None,
+        eval_offload_moe: bool = False,
+        use_cache: bool = True,
+        use_predefined_initial_state: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        # max_position_embeddings is often used to determine the max length
+        # during inference, but samba should have extrapolation abilities
+        self.max_position_embeddings = max(10 * 1024 * 1024,
+                                           max_position_embeddings)
+        self.hidden_size = hidden_size
+        self.rms_norm_eps = rms_norm_eps
+
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.hidden_size_per_head = hidden_size_per_head
+        self.num_key_value_heads = num_key_value_heads
+        self.attention_window_size = attention_window_size
+
+        self.mamba_d_state = mamba_d_state
+        self.mamba_d_conv = mamba_d_conv
+        self.mamba_num_heads = mamba_num_heads
+        self.mamba_step = mamba_step
+        self.mamba_chunk_size = mamba_chunk_size
+
+        self.intermediate_size = intermediate_size
+
+        self.vocab_size = vocab_size
+
+        self.n_expert = n_expert
+        self.k_expert = k_expert
+        self.sparse_intermediate_size = sparse_intermediate_size
+        self.shared_intermediate_size = shared_intermediate_size
+        self.expert_dropout = expert_dropout
+        self.capacity_factor = capacity_factor
+        self.group_size = group_size
+        self.sparse_step = sparse_step
+
+        self.linear_type = linear_type
+        self.fp8_accum_dtype = fp8_accum_dtype
+
+        self.eval_attention_n_bit = eval_attention_n_bit
+        self.eval_mlp_n_bit = eval_mlp_n_bit
+        self.eval_offload_moe = eval_offload_moe
+        self.use_cache = use_cache
+
+        self.use_predefined_initial_state = use_predefined_initial_state
+
+        super().__init__(
+            tokenizer_class=tokenizer_class,
+            pad_token_id=pad_token_id,
+            bos_token_id=bos_token_id,
+            eos_token_id=eos_token_id,
+            tie_word_embeddings=tie_word_embeddings,
+            **kwargs,
+        )
+
+
+class PlamoPreTrainedModel(PreTrainedModel):  # type: ignore
+    config_class = PlamoConfig
+    _no_split_modules: List[str]
+    base_model_prefix = "model"
+    supports_gradient_checkpointing = True
+    _no_split_modules = ["PlamoDecoderLayer"]
+    _skip_keys_device_placement = "past_key_values"
+    _keys_to_ignore_on_load_unexpected = [r"decoder\.version"]
```

Comment:
These all used unused. In general it's best to not check in unused code (engineers should assume any unused, unexercised code is broken), so could you generally go through this file and clean up anything unused?